alea/include/alps/alea/transform.hpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 1998-2017 ALPS Collaboration. See COPYRIGHT.TXT
+ * All rights reserved. Use is subject to license terms. See LICENSE.TXT
+ * For use in publications, see ACKNOWLEDGE.TXT
+ */
+#pragma once
+
+#include <alps/alea/core.hpp>
+#include <alps/alea/computed.hpp>
+
+#include <alps/alea/mean.hpp>
+#include <alps/alea/propagation.hpp>
+
+#include <random>
+
+namespace alps { namespace alea {
+
+/**
+ * Linear transformation mediated by a matrix.
+ */
+template <typename T>
+struct linear_transform
+    : public transform<T>
+{
+public:
+    template <typename Derived>
+    linear_transform(const Eigen::MatrixBase<Derived> &mat)
+        : mat_(mat)
+    { }
+
+    size_t in_size() const { return mat_.rows(); }
+
+    size_t out_size() const { return mat_.cols(); }
+
+    column<T> operator() (const column<T> &in) const
+    {
+        // TODO figure this out
+        return mat_ * typename eigen<T>::col(in);
+    }
+
+    bool is_linear() const { return true; }
+
+private:
+    typename eigen<T>::matrix mat_;
+};
+
+// TODO: make this more general
+
+template <typename T>
+struct scalar_unary_transform
+{
+public:
+    scalar_unary_transform(const std::function<T(T)> &fn) : fn_(fn) { }
+
+    size_t in_size() const { return 1; }
+
+    size_t out_size() const { return 1; }
+
+    column<T> operator() (const column<T> &in) const
+    {
+        if (in.size() != in_size())
+            throw size_mismatch();
+
+        column<T> ret(1);
+        ret(0) = fn_(in(0));
```

Comment:
Can you elaborate on that? I don't know `std::initializer_list` so well . . .

Reply:
Here is an example of how `std::initializer_list` is used to make user-defined classes list-initializable.
```c++
#include <iostream>
#include <vector>
#include <initializer_list>

struct my_struct {
  
  std::vector<double> data;
  
  // Enable list-initialization for my_struct
  my_struct(std::initializer_list<double> l) :
    data(l.begin(), l.end()) {}    
};

int main()
{
    // List-initialization at work
    my_struct ms1 = {1.0, 2.0, 3.0};
    for(double x : ms1.data) std::cout << x << std::endl;

    // List-initialization at work, version 2
    my_struct ms2{4.0, 5.0, 6.0};
    for(double x : ms2.data) std::cout << x << std::endl;
    
    return 0; 
}
```

Reply:
With list-initialization you would be able to write
```c++
    return column<T>{fn_(in(0), in(1))};
```
instead of
```c++
    column<T> ret(1);
    ret(0) = fn_(in(0), in(1));
    return ret;
```